app/components/file_attachment_list/file_attachment_document.js

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Alert,
+    Platform,
+    StyleSheet,
+    TouchableOpacity,
+    View
+} from 'react-native';
+import OpenFile from 'react-native-doc-viewer';
+import RNFetchBlob from 'react-native-fetch-blob';
+import {AnimatedCircularProgress} from 'react-native-circular-progress';
+import {intlShape} from 'react-intl';
+
+import {changeOpacity} from 'app/utils/theme';
+import {getFileUrl} from 'mattermost-redux/utils/file_utils.js';
+
+import {DeviceTypes} from 'app/constants/';
+
+import FileAttachmentIcon from './file_attachment_icon';
+
+const {DOCUMENTS_PATH} = DeviceTypes;
+export const SUPPORTED_DOCS_FORMAT = [
+    'application/pdf',
+    'application/msword',
+    'application/vnd.ms-excel',
+    'application/vnd.ms-powerpoint',
+    'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'application/xml',
+    'text/csv'
+];
+
+export default class FileAttachmentDocument extends PureComponent {
+    static propTypes = {
+        iconHeight: PropTypes.number,
+        iconWidth: PropTypes.number,
+        file: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        wrapperHeight: PropTypes.number,
+        wrapperWidth: PropTypes.number
+    };
+
+    static defaultProps = {
+        iconHeight: 65,
+        iconWidth: 65,
+        wrapperHeight: 100,
+        wrapperWidth: 100
+    };
+
+    static contextTypes = {
+        intl: intlShape
+    };
+
+    state = {
+        didCancel: false,
+        downloading: false,
+        progress: 0
+    };
+
+    componentDidMount() {
+        this.mounted = true;
+    }
+
+    componentWillUnmount() {
+        this.mounted = false;
+    }
+
+    cancelDownload = () => {
+        if (this.mounted) {
+            this.setState({didCancel: true});
+        }
+
+        if (this.downloadTask) {
+            this.downloadTask.cancel();
+        }
+    };
+
+    downloadAndPreviewFile = async (file) => {
+        const path = `${DOCUMENTS_PATH}/${file.name}`;
+
+        this.setState({didCancel: false});
+
+        try {
+            const isDir = await RNFetchBlob.fs.isDir(DOCUMENTS_PATH);
+            if (!isDir) {
+                try {
+                    await RNFetchBlob.fs.mkdir(DOCUMENTS_PATH);
+                } catch (error) {
+                    // Do nothing
```

Comment:
What happens when the mkdir fails? This looks like a place where we can't just do nothing when the folder can't be created

Reply:
I can fix this and show and error message